tasks/snyk-code-test.yaml

```diff
@@ -0,0 +1,59 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: snyk-code-test
+spec:
+  description: >-
+    Static code security test with snyk
+  results:
+    - description: Test output
+      name: HACBS_TEST_OUTPUT
+  params:
+    - name: shared-secret
+      default: test-team-snyk
+    - name: args
+      type: string
+      description: extra args needs to append
+      default: "--all-projects --exclude=test*,vendor,deps"
+  volumes:
+    - name: snyk-secret
+      csi:
+        readOnly: true
+        driver: csi.sharedresource.openshift.io
+        volumeAttributes:
+          sharedSecret: $(params.shared-secret)
+  steps:
+    - name: snyk-code-test
+      image: quay.io/redhat-appstudio/hacbs-test:feature-sast
+      workingDir: $(workspaces.workspace.path)
+      volumeMounts:
+        - name: snyk-secret
+          mountPath: "/etc/secrets"
+          readOnly: true
+      script: |
+        #!/usr/bin/env bash
+        SNYK_TOKEN="$(cat /etc/secrets/snyk_token)"
+        if [[ -z $SNYK_TOKEN ]]; then
+          echo "SNYK_TOKEN is empty and a secret 'test-team-snyk' which includes 'snyk_token' need to be created in test-team namespace"
+          exit 0
+        fi
+        export SNYK_TOKEN
+        SNYK_EXIT_CODE=0
+        mkdir test
```

Comment:
`test` is very generic, so here is high risk that the folder will exist in the source code (that's in this workspace). Rather use `$(workspaces.workspace.path)/hacbs/$(context.task.name)` as in other tasks.

Reply:
@Michkov , can you please help to review it again? Thanks